test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs

```diff
@@ -471,7 +497,34 @@ public void Parse_ComplexSegment_OptionalParameterFollowingPeriod_PeriodAfterSla
             Assert.Equal<RouteTemplate>(expected, actual, new TemplateEqualityComparer());
         }
 
-        [Theory]        
+        [Theory]
+        [InlineData(@"{p1:regex(^\d{{3}}-\d{{3}}-\d{{4}}}$)}")] // extra }
+        [InlineData(@"{p1:regex(^\d{{3}}-\d{{3}}-\d{{4}}$)}}")] // extra } at the end
+        [InlineData(@"{{p1:regex(^\d{{3}}-\d{{3}}-\d{{4}}$)}")] // extra { at the begining
+        [InlineData(@"{p1:regex(([}])\w+}")] // Not escaped }
+        [InlineData(@"{p1:regex(^\d{{3}}-\d{{3}}-\d{{4}$)}")] // Not escaped }
```

Comment:
I am guessing it wont fail in parsing and creating inline constraint, I will fail when creating constraint out of inline constraint, I will check.